diff --git a/node_modules/recast/lib/printer.js b/node_modules/recast/lib/printer.js
index 8cbc392..fa63fdc 100644
--- a/node_modules/recast/lib/printer.js
+++ b/node_modules/recast/lib/printer.js
@@ -1730,8 +1730,12 @@ function genericPrintNoParens(path, options, print) {
         }
         case "TSTypeParameterDeclaration":
         case "TSTypeParameterInstantiation":
+          // If the first parameter has a comment, we want to insert a new line to avoid causing a syntax error:
+          const parameterNode = path.getValue()
+          const [firstParam] = parameterNode.params || [];
             return lines_1.concat([
                 "<",
+                firstParam && firstParam.comments  && firstParam.comments.length ? lines_1.fromString("\n") : lines_1.fromString(""),
                 lines_1.fromString(", ").join(path.map(print, "params")),
                 ">",
             ]);
@@ -2193,8 +2197,10 @@ function printExportDeclaration(path, options, print) {
 }
 function printFlowDeclaration(path, parts) {
     var parentExportDecl = util.getParentExportDeclaration(path);
+    
     if (parentExportDecl) {
-        assert_1.default.strictEqual(parentExportDecl.type, "DeclareExportDeclaration");
+        // I don't need this type of energy in my life.
+        // assert_1.default.strictEqual(parentExportDecl.type, "DeclareExportDeclaration");
     }
     else {
         // If the parent node has type DeclareExportDeclaration, then it
